src/hamiltonian.jl

```diff
@@ -0,0 +1,25 @@
+using OrdinaryDiffEq, ForwardDiff
+
+function HamiltonianProblem(H,q0::Real,p0::Real,tspan::Tuple{T,T};kwargs...) where T<:Real
+    @inline function dq(t,x,v,dx)
+        dx[1] = ForwardDiff.derivative(v->H(x[1], v), v[1])
+    end
+    @inline function dp(t,x,v,dv)
+        dv[1] = ForwardDiff.derivative(x->-H(x, v[1]), x[1])
+    end
+
+    return ODEProblem((dq,dp), ([q0],[p0]), tspan; kwargs...)
```

Comment:
Write this not as vectors. Instead we should just add the out-of-place form for symplectic.